include/openmc/openmp_interface.h

```diff
@@ -0,0 +1,77 @@
+#ifndef OPENMC_OPENMP_INTERFACE_H
+#define OPENMC_OPENMP_INTERFACE_H
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+namespace openmc {
+
+//==============================================================================
+//! An object used to prevent concurrent access to a piece of data.
+//
+//! This type meets the C++ "Lockable" requirements.
+//==============================================================================
+
+class ThreadMutex
```

Comment:
I would also probably rename this `OMPThreadMutex` to be a bit more specific (since there are other kinds of C++ threads).